src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/CaseWhenParser.java

```diff
@@ -45,6 +48,15 @@ public CaseWhenParser(SQLCaseExpr caseExpr, String alias, String tableAlias) {
     public String parse() throws SqlParseException {
         List<String> result = new ArrayList<>();
 
+        if (caseExpr.getValueExpr() instanceof SQLIdentifierExpr) {
```

Comment:
what if caseExpr.getValueExpr() instanceof SQLIdentifierExpr return false?

Reply:
> what if caseExpr.getValueExpr() instanceof SQLIdentifierExpr return false?

If we do support for example `CASE WHEN LOG(age) = 1 ...`, maybe we can do this in negate way, `if valueExpr is not instanceof SQLBinaryOp` ?

Reply:
Typically the (SQLCaseExpr) caseExpr has a couple of items and a valueExpr. Each item gets a left expr, a right expr, and a binary operator expr, these 3 parts come to the conditionExpr. And valueExpr is null.  So it will go directly to traverse the items under the caseExpr.

But in this grammer `select case weather when 'sunny' then ...` the parser recognizes `weather` as the (SQLIdentifierExpr) valueExpr, but in the item the conditionExpr is kind of in a mess.